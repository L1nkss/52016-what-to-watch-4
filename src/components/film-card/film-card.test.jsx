import FilmCard from "@components/film-card/film-card";

// Моки
const film = {
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

const cb = () => {};

describe(`Testing FilmCard component`, () => {
  it(`Component should successfully rendered`, () => {
    const component = renderer.create(
        <FilmCard
          filmInfo={film}
          renderPlayer={cb}
          onHoverCardHandler = {cb}
          handleMouseEnter = {cb}
          handleMouseLeave = {cb}
          onClickCardHandler = {cb}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
