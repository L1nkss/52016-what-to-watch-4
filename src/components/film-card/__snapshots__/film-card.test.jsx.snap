// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilmCard component Component should successfully rendered 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <a
    className="c-inherit"
    href="/film-detail/2"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    />
    <h3
      className="small-movie-card__title"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </h3>
  </a>
</article>
`;
